ceno_emul/src/rv32im_encode.rs

```diff
@@ -49,7 +49,8 @@ const fn encode_i(kind: InsnKind, rs1: u32, rd: u32, imm: u32) -> u32 {
     let rd = rd & MASK_5_BITS;
     let func3 = kind.codes().func3;
     let opcode = kind.codes().opcode;
-    let imm = imm & MASK_12_BITS;
+    let shift_func = (matches!(kind, InsnKind::SRAI) as u32) << 10; // SRAI kind is encoded in imm.
```

Comment:
Could we elaborate more on this operation? It looks to me like an special encoding outside of riscv spec

Reply:
Adding a comment might be useful, but as far as I can tell, it's pretty much in line with the spec.

See section 2.4.1 of the [spec](https://github.com/riscv/riscv-isa-manual/releases/download/20240411/unpriv-isa-asciidoc.pdf).

Reply:
I see, thanks for the reference

Then it might make thing clear
```diff
diff --git a/ceno_emul/src/rv32im_encode.rs b/ceno_emul/src/rv32im_encode.rs
index 03128c27..61eed9fe 100644
--- a/ceno_emul/src/rv32im_encode.rs
+++ b/ceno_emul/src/rv32im_encode.rs
@@ -49,8 +49,8 @@ const fn encode_i(kind: InsnKind, rs1: u32, rd: u32, imm: u32) -> u32 {
     let rd = rd & MASK_5_BITS;
     let func3 = kind.codes().func3;
     let opcode = kind.codes().opcode;
-    let shift_func = (matches!(kind, InsnKind::SRAI) as u32) << 10; // SRAI kind is encoded in imm.
-    let imm = imm & MASK_12_BITS | shift_func;
+    let is_arithmetic_right_shift = (matches!(kind, InsnKind::SRAI) as u32) << 10; // SRLI/SRAI kind is encoded in imm[10].
+    let imm = imm & MASK_12_BITS | is_arithmetic_right_shift;
     imm << 20 | rs1 << 15 | func3 << 12 | rd << 7 | opcode
 }
```

Reply:
The rename looks good to me.  One cosmetic suggestion: put `// SRLI/SRAI kind is encoded in imm[10].` before the line, not behind it.

Reply:
done